gcovr/coverage.py

```diff
@@ -197,8 +260,10 @@ def update(self, other):
         for branch_id, branch_cov in other.branches.items():
             self.branch(branch_id).update(branch_cov)
 
-        for decision_id, decision_cov in other.decisions.items():
-            self.decision(decision_id).update(decision_cov)
+        if self.decision is None:
+            self.decision = other.decision
+        else:
+            self.decision.update(other.decision)
```

Comment:
This update could fail for conflicting decision types, but I don't think that can occur in practice. So I'm fine with this for now. Not sure what an elegant solution would be.

Reply:
That can occure if the preprocessor is used and different code is generated but I think for the moment I've no solution for this.